internal/reader/home_chain.go

```diff
@@ -115,24 +118,42 @@ func (r *homeChainPoller) poll() {
 }
 
 func (r *homeChainPoller) fetchAndSetConfigs(ctx context.Context) error {
-	var chainConfigInfos []ChainConfigInfo
-	err := r.homeChainReader.GetLatestValue(
-		ctx,
-		consts.ContractNameCCIPConfig,
-		consts.MethodNameGetAllChainConfigs,
-		primitives.Unconfirmed,
-		nil,
-		&chainConfigInfos,
-	)
-	if err != nil {
-		return err
+	var allChainConfigInfos []ChainConfigInfo
+	pageIndex := uint64(0)
+
+	for {
+		var chainConfigInfos []ChainConfigInfo
```

Comment:
Can we add something like the following for extra safety?

```
const pageIndexHardLimit = 100

for {
   ...
   pageIndex++

   if pageIndex > pageIndexHardLimit {
      return fmt.Errorf("something is wrong!")
   }
}
```

Reply:
added extra safety with hardCap for pageIndex
its included in for loop and once for loop exits we check if the pageIndex > pageIndexHardLimit
log a warn message

but wouldnt this cap the max chains we can extract the config from?

@dimkouv

Reply:
We can add some big value for the limit.

Reply:
Why do we need a hard limit, as opposed to just giving a warning and fetching everything?

Reply:
I think it would be painful to debug if we don't return all pages and just emit a warning (as opposed to either returning all pages, or throwing an exception)